### ISSUE TITLE
Evaluate exposure to or impact of reported Thrift vulnerabilities

### ISSUE THREAD

**Issue by mike-myers-tob** [osquery/osquery#7104]
# Task

Apache Thrift 0.14.1 is the latest release, but there is apparently a list of vulnerabilities that they may not have fixed.

User-submitted report generated by Black Duck:

<img width="1344" alt="Screen Shot 2021-05-12 at 4 04 05 PM" src="https://user-images.githubusercontent.com/30631532/118170107-9347cd00-b3de-11eb-82ff-de4b56485c36.png">

In discussion so far, the point was made that osquery's Thrift server only accepts connections from local processes running as the root or Administrator user.

**Comment by marcosd4h** [comment]
Thrift use named pipes on Windows as the IPC communication mechanism. I've just checked the security descriptor of the named pipe used by Thrift and permissions are the expected ones (the named pipe is only accessible by Administrators and SYSTEM as it was set by [#6875](https://github.com/osquery/osquery/pull/6875))

Below is a screenshot that gives more details on this. On the right, you can see the named pipe handle from `osqueryd.exe` process. On the left, you can see the [SDDL](https://learn.microsoft.com/en-us/windows/win32/secauthz/security-descriptor-definition-language) representation of the security descriptor from Thrift's named pipe `\\.\pipe\osquery.em`. The SDDL representation effectively shows that only `BUILTIN\Administrators` and `NT AUTHORITY\SYSTEM` have rights on the named pipe

![image](https://user-images.githubusercontent.com/17311749/197872036-30de9e02-bdba-48b6-bec6-2112b551371e.png)

**Comment by mike-myers-tob** [comment]
Yes, the Thrift channel is secured against use by non-Administrators. That was true previously for POSIX systems, and then true for Windows after I fixed it in the PR that Seph referred to above. So our exposure would seem to be minimal, even if Thrift had a critical vulnerability. I think this is enough evaluation, really, since we state this in our [ASSURANCE.md documentation](https://github.com/osquery/osquery/blob/master/ASSURANCE.md#daemon-runs-as-root-or-administrator).

For what it's worth, we link with Apache Thrift 0.15 now. I just checked. It could be updated, since version 0.17.0 is now the latest.

Regarding these individual vulnerabilities, Black Duck is doing its usual false positive thing and detecting `thrift.dll` as being the Facebook version, whereas we use the Apache fork. I have no idea if any/all of the bugs found in the Facebook version are also present in the Apache version, and this would be very expensive timewise to individually investigate. The [NVD entry for CVE-2021-24028 for instance](https://nvd.nist.gov/vuln/detail/CVE-2021-24028), only indicates Facebook Thrift. Our new CVE monitoring system that @Smjert set up has not generated any issues related to our Thrift dependency, probably because NVD doesn't list them as problems in the Apache Thrift, and the automation is using NVD as the vulnerability authority.
